zip_0244.py

```diff
@@ -45,6 +51,20 @@ def sapling_digest(tx):
 
     return digest.digest()
 
+def sapling_auth_digest(tx):
+    digest = blake2b(digest_size=32, person=b'ZTxAuthSapliHash')
+
+    if len(tx.vSpendsSapling) + len(tx.vOutputsSapling) > 0:
+        for desc in tx.vSpendsSapling:
+            digest.update(bytes(desc.proof))
```

Comment:
Isn't `desc.proof` already a bytes value?

Reply:
Not for Sapling (we have a `GrothProof` type).